apps/sms/views/conversation/js/attachment.js

```diff
@@ -26,8 +26,7 @@
   function Attachment(blob, options) {
     options = options || {};
     this.blob = blob;
-    this.name = blob.name || options.name ||
-      navigator.mozL10n.get('unnamed-attachment');
+    this.name = blob.name || options.name || 'undefined';
```

Comment:
Why not, but I think some code would break if this.name is not a string; so let's at least assign an empty string in the constructor.

* For the first case (SMIL generation) I think it's correct if the name stays blank.
* for the gallery viewing, I think we need to show something better, but this should be handled by the gallery app instead (maybe they do it already).
* for the rendering inside the SMS app, I think you're right, and this could be done in attachment_renderer.
* for the last case, we don't really care but we need at least a string.

This is enough of a edge case that it could be handled in a follow-up, but if you want to do this here I'd be happy :)